Description: gdm_slave_get_timed_login_details() does not fill variables if username is NULL
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=596830
Bug-Ubuntu: https://bugs.edge.launchpad.net/ubuntu/+source/gdm/+bug/435801

Index: gdm-2.32.1/daemon/gdm-simple-slave.c
===================================================================
--- gdm-2.32.1.orig/daemon/gdm-simple-slave.c	2011-03-29 03:48:50.000000000 +1100
+++ gdm-2.32.1/daemon/gdm-simple-slave.c	2011-04-08 16:32:16.451921308 +1000
@@ -1164,6 +1164,7 @@
         if (res) {
                 gboolean enabled;
                 int      delay;
+                char    *username;
 
                 /* FIXME: handle wait-for-go */
 
@@ -1171,7 +1172,8 @@
 
                 delay = 0;
                 enabled = FALSE;
-                gdm_slave_get_timed_login_details (GDM_SLAVE (slave), &enabled, NULL, &delay);
+                gdm_slave_get_timed_login_details (GDM_SLAVE (slave), &enabled, &username, &delay);
+                g_free (username);
                 if (! enabled || delay > 0) {
                         start_greeter (slave);
                         create_new_session (slave);
